docs/madr/decisions/074-naming-internal-envoy-resources.md

```diff
@@ -0,0 +1,152 @@
+# Standardized Naming for internal xDS Resources
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/13266
+
+## Context and Problem Statement
+
+We use resource names to identify listeners, clusters, routes, virtual hosts and secrets in the xDS APIs.
+These names appear in metrics, logs, and debug endpoints (config_dump), and they are often used for monitoring, filtering, and diagnostics.
+
+It's important to know the difference between internal resources (the ones that are defined by Kuma itself, and they usually do not concern the end users) and external resources (the ones that are defined indirectly by the user and the users are interested in).
+For example, clusters like `kuma:envoy:admin` or `_kuma:dynamicconfig` are internal to Kuma, while `kri_msvc_mesh-1_us-east-2_kuma-demo_backend_httpport` cluster is an external resource.
+
+Currently, there is inconsistency in how these resource names are formed, some of them have a `_kuma` prefix, some of them just contain `kuma` and some are free form.
+
+Some internal resources do not map clearly to Kuma resources in the data store, making it difficult to relate metrics to Kuma abstractions.
+
+There are also resources that are in a bit of a gray area, like `meshtrace_datadog` which is a cluster created for `MeshTrace` policy.
+It seems internal, but it could be traced down to MeshTrace resource `kri_meshtrace_mesh-1__kuma-system_my-meshtrace_`.
+
+#### Examples of kuma resource names
```

Comment:
It looks like a full list of internal resources, or some internal resources are missing? If it's full list then I'd replace the word "Examples" and call it "Internal Kuma resources". It makes it sound like a list you can work on

Reply:
I did not find more but you know how it is - there could be something dynamically defined that is triggered by some specific setting.